dspace-api/src/main/java/org/dspace/importer/external/ris/service/RisImportMetadataSourceServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.importer.external.ris.service;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Resource;
+
+import org.dspace.importer.external.exception.FileSourceException;
+import org.dspace.importer.external.service.components.AbstractPlainMetadataSource;
+import org.dspace.importer.external.service.components.dto.PlainMetadataKeyValueItem;
+import org.dspace.importer.external.service.components.dto.PlainMetadataSourceDto;
+
+/**
+ * Implements a metadata importer for RIS files
+ * Implementations insprider by BTE DataLoader {@link https://github.com/EKT/Biblio-Transformation-Engine/blob/master/bte-io/src/main/java/gr/ekt/bteio/loaders/RISDataLoader.java}
+ * 
+ * @author Pasquale Cavallo (pasquale.cavallo at 4science dot it)
+ */
+public class RisImportMetadataSourceServiceImpl extends AbstractPlainMetadataSource {
+
+    @Override
+    public String getImportSource() {
+        return "RISMetadataSource";
+    }
+
+    protected List<PlainMetadataSourceDto> readData(InputStream inputStream) throws FileSourceException {
+        return aggreageteData(inputStream);
+    }
+
+    private List<PlainMetadataSourceDto> aggreageteData(InputStream inputStream) throws FileSourceException {
```

Comment:
Minor spelling typo.  This should be named `aggregateData`.  As this is an important private method, I'd like to see some JavaDocs (or a basic comment) to describe what this method is doing to "aggregate" the data.